pkg/controller/shared_state.go

```diff
@@ -1564,3 +1571,60 @@ func (c *SharedState) getNodesTopologySelector(ctx context.Context, nodes []v1.N
 	}
 	return topologySelector
 }
+
+func (s *SharedState) addNodeToAvailableAttachmentsMap(ctx context.Context, node string) {
+	nodeObj := &v1.Node{}
+	if err := s.cachedClient.Get(ctx, types.NamespacedName{Name: node}, nodeObj); err != nil {
+		capacity, _ := azureutils.GetNodeMaxDiskCountWithLabels(nodeObj.Labels)
+		var count atomic.Int32
+		count.Store(int32(capacity))
+		s.availableAttachmentsMap.Store(node, &count)
+		return
+	}
+	capacity, _ := azureutils.GetNodeRemainingDiskCount(ctx, s.cachedClient, node)
+	var count atomic.Int32
+	count.Store(int32(capacity))
+	s.availableAttachmentsMap.Store(node, &count)
+}
+
+func (s *SharedState) deleteNodeFromAvailableAttachmentsMap(ctx context.Context, node string) {
+	w, _ := workflow.GetWorkflowFromContext(ctx)
```

Comment:
`deleteNodeFromAvailableAttachmentMap` is mostly called from the `Reconcile` function which doesn't create a workflow. Relying on the temporary workflow that will be returned by `GetWorkflowFromContext` for logging purposes is overkill. You can just use `klog` directly.

Reply:
Done.